### PR TITLE
chore: merge `v3.18.0-emqx` into `v3.19.0-emqx`

### PR DESCRIPTION
**Pull request by keynslug**
Version v3.19.0 was picked as the first one to support OTP 25.x officially. The latest is v3.20.0 but it changes the project structure considerably and will require a bit more effort to merge all the patches.

### CODE REVIEW

File: src/rebar.app.src.script [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 {application, rebar,
  [{description, "Rebar: Erlang Build Tool"},
-  {vsn, "3.19.0"},
+  {vsn, {cmd, "git describe --tags"}},
```
Comment (ieQu1) [inline src/rebar.app.src.script:6]: `{vsn, git}` has same effect?

Comment (keynslug) [reply src/rebar.app.src.script:6]: Yes and no. Same when HEAD is tagged with some tag but different when HEAD is ahead of some tag where rebar [ensures that vsn is semver-valid](https://github.com/emqx/rebar3/blob/50c7ebca4d7fdc6a8963eb31613d9ffcfc090b12/src/rebar_git_resource.erl#L352-L358). Nevertheless, `{vsn, git}` should work better here.

Comment (keynslug) [reply src/rebar.app.src.script:6]: Ah, looks like `{vsn, git}` isn't available during bootstrap. 😓 
```
vcs_vsn: Unknown vsn format: git
```

Comment (zmstone) [reply src/rebar.app.src.script:6]: yeah, that is why `cmd` is used here.

File: build [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        halt(1)
+end.
+EOF
+)"
+
+erl -noshell -eval "${CHECK_VSN}"
+
+if [ "${1:-}" = "--in-docker" ]; then
+    #shellcheck disable=SC2086
+    exec docker run $maybeit --rm -v "$(pwd):/src" --workdir /src erlang:25.1.2 ./build
```
Comment (zmstone) [inline build]:
```suggestion
    exec docker run $maybeit --rm -v "$(pwd):/src" --workdir /src erlang:24 ./build
```

**Review by zmstone — APPROVED** [review]

**Pull request merged**